### PR TITLE
Update deprecated commands

### PR DESCRIPTION
**Pull request by Hajara-I**
### What is the context of this PR?
Set-output command refactored. CodeQL action updated to v2. Git checkout HEAD^2 removed from workflow. Python updated to v4. Ubuntu updated to v22. Actions/checkout, actions/cache, actions/setup-node updated to v3. Set python dependencies in codeQL as false. 

### How to review 
Create a draft PR to ensure that all warnings have been resolved and no outstanding warnings remain.

### CODE REVIEW

**Comment by berroar** [comment]
[Fix warnings for deprecated commands in Github Actions (2 SP)](https://trello.com/c/WIO9ePeB/5462-fix-warnings-for-deprecated-commands-in-github-actions-2-sp)

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -36,25 +36,20 @@ jobs:
         # a pull request then we can checkout the head.
         fetch-depth: 2
 
-    # If this run was triggered by a pull request event, then checkout
```
Comment (berroar) [inline .github/workflows/codeql-analysis.yml:39]: How come this bit was removed>

Comment (Hajara-I) [reply .github/workflows/codeql-analysis.yml:39]: It was due to a warning stating that: git checkout HEAD^2 is no longer necessary. See here https://github.com/ONSdigital/eq-questionnaire-runner/actions/runs/3839459914

File: .github/workflows/pull_request.yml [hunk 2]
```diff
@@ -40,13 +40,13 @@ jobs:
   node-dependencies:
     runs-on: ubuntu-18.04
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions/setup-node@v1
+      - uses: actions/checkout@v3
```
[the remark this answers is not in the document]
Comment (Hajara-I) [reply .github/workflows/pull_request.yml]: Node changes have been reverted.

**Review by berroar — COMMENTED** [review]
Probably worth updating the PR description as it's not only set-output that has been updated now 👍

**Review by Yuyuutsu — APPROVED** [review]
Tested it on a Draft Pull requests, deprecated command warnings are gone except for Node, but that is on another card. 👍

**Review by Yuyuutsu — APPROVED** [review]

File: .github/workflows/pull_request.yml [hunk 3]
```diff
@@ -112,28 +112,28 @@ jobs:
...
       - name: Update to latest stable Chrome
         run: |
           wget https://dl.google.com/linux/direct/google-chrome-stable_current_amd64.deb
           sudo apt install ./google-chrome-stable_current_amd64.deb
-      - uses: actions/setup-node@v1
+      - uses: actions/setup-node@v3
         with:
-          node-version: "14.19.0"
+          node-version: "14"
```
Comment (creativeappdevamit) [inline .github/workflows/pull_request.yml:126]: Hi, Just wanted to check
The earlier version had specific version(14.9.0), here we have changed it to Major Version (14). Any specific reason for this change?

File: .github/workflows/pull_request.yml [hunk 4]
```diff
@@ -30,26 +30,26 @@ jobs:
...
-    runs-on: ubuntu-18.04
+    runs-on: ubuntu-20.04
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions/setup-node@v1
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
         with:
-          node-version: "14.19.0"
+          node-version: "14"
```
Comment (creativeappdevamit) [inline .github/workflows/pull_request.yml:46]: Hi, Just wanted to check
The earlier version had specific version(14.9.0), here we have changed it to Major Version (14). Any specific reason for this change?

File: .github/workflows/pull_request.yml (hunk 3 above)
Comment (Hajara-I) [reply .github/workflows/pull_request.yml]: Hi, I have refactored this back to version 14.19.0. Cheers.

File: .github/workflows/pull_request.yml (hunk 4 above)
Comment (Hajara-I) [reply .github/workflows/pull_request.yml]: Hi, I have refactored this back to version 14.19.0. Cheers.

**Review by creativeappdevamit — APPROVED** [review]
LGTM

**Pull request merged**